processor/groupbyattrsprocessor/processor.go

```diff
@@ -28,13 +28,111 @@ type groupbyattrsprocessor struct {
 
 // ProcessTraces process traces and groups traces by attribute.
 func (gap *groupbyattrsprocessor) ProcessTraces(_ context.Context, td pdata.Traces) (pdata.Traces, error) {
-	// TODO
+	rss := td.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+
+		extractedResourceSpans := newSpanAttributeGroups()
+
+		ilss := rs.InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			for k := 0; k < ils.Spans().Len(); k++ {
+				span := ils.Spans().At(k)
+
+				groupedAnything, groupedAttrMap, nonGroupedAttrMap := gap.splitAttrMap(span.Attributes())
+				if groupedAnything {
+					mNumGroupedSpans.M(1)
+					overwriteAttributes(nonGroupedAttrMap, span.Attributes())
```

Comment:
Does this processor really need to overwrite non-grouped attributes?

Reply:
Changed the flow to delete unnecessary attributes as suggested in another comment